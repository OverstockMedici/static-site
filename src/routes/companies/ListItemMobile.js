import React from 'react';

const ListItemMobile = ({
    link,
    name,
    image,
    color,
    description,
    linkText,
    openCompanyName,
    cardStyle,
    toggleDescriptionOpen,
}) => {
    return (
        <div
            data-testid="list-mobile"
            className={cardStyle}
            onClick={() => toggleDescriptionOpen(name)}
        >
            <img src={image} alt={`${name} logo`} />
            <div
                data-testid="list-mobile-style"
                className="company-description"
                style={{ backgroundColor: `${color}` }}
            >
                {description} Visit{' '}
                <a href={link} target="_blank" rel="noopener noreferrer">
                    {linkText}
                </a>
            </div>
        </div>
    );
};

export default ListItemMobile;
