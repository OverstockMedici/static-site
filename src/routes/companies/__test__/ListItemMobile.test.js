import React from 'react';
import ReactDOM from 'react-dom';
import ListItemMobile from './../ListItemMobile';

import { render, cleanup } from '@testing-library/react';
import { getByAltText } from '@testing-library/dom';

import renderer from 'react-test-renderer';

afterEach(cleanup);

const listItemMobileProps = {
    link: 'https://www.spacex.com/',
    name: 'SpaceX',
    image: 'https://www.spacex.com/sites/spacex/files/spacex_logo_white.png',
    color: 'rgb(0, 0, 0)',
    description:
        'The first company to achieve orbit with liquid-fueled rocket, SpaceX develops launch systems at dramatically-lower costs than any entity - government or private - in history',
};
const { link, name, image, color, description, linkText } = listItemMobileProps;

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListItemMobile />, div);
});

it('renders List correctly', () => {
    const { getByTestId, container } = render(
        <ListItemMobile {...{ link, name, image, color, description, linkText }} />,
    );

    const imgMetaData = getByAltText(container, 'SpaceX logo');
    const listComponentMetaData = getByTestId('list-mobile-style');

    expect(imgMetaData.alt).toBe('SpaceX logo');
    expect(listComponentMetaData.style.backgroundColor).toBe(color);
    expect(getByTestId('list-mobile')).toHaveTextContent(description);
});

it('matches snapshot', () => {
    const listItemMobileTree = renderer
        .create(
            <ListItemMobile
                {...{ link, name, image, color, description, linkText }}
            />,
        )
        .toJSON();

    expect(listItemMobileTree).toMatchSnapshot();
});
