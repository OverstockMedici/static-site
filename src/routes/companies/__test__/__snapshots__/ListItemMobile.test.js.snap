// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="list-mobile"
  onClick={[Function]}
>
  <img
    alt="SpaceX logo"
    src="https://www.spacex.com/sites/spacex/files/spacex_logo_white.png"
  />
  <div
    className="company-description"
    data-testid="list-mobile-style"
    style={
      Object {
        "backgroundColor": "rgb(0, 0, 0)",
      }
    }
  >
    The first company to achieve orbit with liquid-fueled rocket, SpaceX develops launch systems at dramatically-lower costs than any entity - government or private - in history
     Visit
     
    <a
      href="https://www.spacex.com/"
      rel="noopener noreferrer"
      target="_blank"
    />
  </div>
</div>
`;
